### DISCUSSION TITLE
Trouble with LPS Configuration for Loco Positioning Node

### DISCUSSION THREAD

**Discussion by ingersolls** [bitcraze/discussions#1035] [Q&A]
My group is currently working on the crazyflie program for a class project and we are having issues with configuring the Loco Positioning Nodes. We were able to successfully flash each of the nodes with the firmware by following the step by step directions given online. When we go onto the next step to configure each node and label them, the LPS configuration tool does not give us the option to click apply when trying to apply the configuration into each node.

**Comment by knmcguire** [comment]
Hi! So I'm assuming that you already found the tutorial doc and repo doc on this: 

- https://www.bitcraze.io/documentation/tutorials/getting-started-with-loco-positioning-system/#preparing-the-anchors
- https://www.bitcraze.io/documentation/repository/lps-node-firmware/master/

Seems that perhaps it has something to do with the usb access. Are you running the tool on windows, linux or mac?

**Comment by ingersolls** [comment]
We have been running the tool on windows.
________________________________

**Comment by knmcguire** [comment]
So, this is a long known bug for a while and unfortunately we didn't have the time to fix this... I've tried it out myself but the .exe isn't working anymore. Only firmware upgrades but it doesn't recognize the device even with zadig installing the right driver for the loco positioning node. 

 https://github.com/bitcraze/lps-tools/issues/19

I recommend you updating the nodes through linux. If you don't have a linux computer at your disposal, you can install LPS tool on linux through:

- Our ubuntu virtual machine: https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/#inst-virtualmachine
- use usbipd to connect the USB device if you have WSL installed on your windows 11: https://github.com/dorssel/usbipd-win/wiki/WSL-support

**Reply by ingersolls** [reply]
So we downloaded your virtual machine and reattempted to update the nodes using a linux computer. After doing so we tried to connect the positioning nodes to software but we are still having the same connecting issues.

**Reply by knmcguire** [reply]
So, if you have got a linux computer, have you tried to use the tool also directly on the computer as well? that would work much better.

This is also an alternative way to flash the LPS nodes as well: https://www.bitcraze.io/documentation/repository/lps-node-firmware/master/building/build_flash/

Don't forget to look at the building from source instructions for the LPS tools instructions as well: https://github.com/bitcraze/lps-tools

**Comment by ingersolls** [comment]
Okay I will try this later today when I am back in the lab. Thank you!
________________________________

**Comment by ingersolls** [comment]
Hey Kimberly, unfortunately we don't have access to a linux computer in the lab. I will attempt to flash the nodes with the alternative way and let you know of any updates, thanks!
________________________________

**Comment by knmcguire** [comment]
A related discussion that might contain some tips as well: https://github.com/orgs/bitcraze/discussions/1036
